File: 2810-faulty-keyboard/2810-faulty-keyboard.js
/**
 * @param {string} s
 * @return {string}
 */
var finalString = function(s) {
        newString = "";
        for(let char of s){
            if(char=='i') newString = newString.split("").reverse().join("");
            else newString += char;
        }
        return newString;
};

File: 2810-faulty-keyboard/2810-faulty-keyboard.java
class Solution {
    public String finalString(String s) {
        StringBuilder newString = new StringBuilder("");;
        for(int i = 0;i<s.length();i++){
            char c = s.charAt(i);
            if(c=='i') newString.reverse();
            else newString.append(c);
        }
        return newString.toString();
    }
}